File: ios/CounterViewManager.m
//
//  CounterViewManager.m
//  Counter
//
//  Created by Jiweon Park on 2024-02-16.
//

#import <Foundation/Foundation.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(CounterViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(count, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onUpdate, RCTDirectEventBlock)
RCT_EXTERN_METHOD(
  updateFromManager:(nonnull NSNumber *)node
  count:(nonnull NSNumber *)count
)
@end

File: ios/CounterViewManager.swift
//
//  CounterViewManager.swift
//  Counter
//
//  Created by Jiweon Park on 2024-02-16.
//

@objc(CounterViewManager)
class CounterViewManager: RCTViewManager {
  override func view() -> UIView! {
      return CounterView()
    }
  
  override static func requiresMainQueueSetup() -> Bool {
      return true
    }
  
  @objc func updateFromManager(_ node: NSNumber, count: NSNumber) {
    
    DispatchQueue.main.async {                                // 2
      let component = self.bridge.uiManager.view(             // 3
        forReactTag: node                                     // 4
      ) as! CounterView                                       // 5
      component.update(value: count)                          // 6
    }
  }
  
}
